linkerd/proxy/transport/src/addrs.rs

```diff
@@ -28,6 +28,9 @@ pub struct Local<T>(pub T);
 #[derive(Copy, Clone, Debug, Eq, PartialEq, Hash)]
 pub struct Remote<T>(pub T);
 
+#[derive(Eq, PartialEq, Debug, Copy, Clone, Hash)]
+pub struct TargetPort(u16);
+
```

Comment:
Why do we need a new type for this? Isn't this already covered by `Remote<ServerAddr>`?